Core/GDCore/IDE/Events/EventsVariablesFinder.h

```diff
@@ -55,8 +54,8 @@ class EventsVariablesFinder {
    */
   static std::set<gd::String> FindAllLayoutVariables(
       const gd::Platform& platform,
-      const gd::Project& project,
-      const gd::Layout& layout);
+      gd::Project& project,

```

Comment:
Why no const anymore?

Reply:
Ho, yes, I wanted to ask you. Do you think we should make a duplicated implementation of `ArbitraryEventsWorkerWithContext` for read-only ones?

Reply:
That would make sense. We could have a `ReadOnlyArbitraryEventsWorker` and a `ReadOnlyArbitraryEventsWorkerWithContext`. 
Of course, it means doubling the implementation - which is not super great (and at the same time not too much code...).

You could either do that or choose to implement ReadOnlyArbitraryEventsWorker by reusing `ArbitraryEventsWorker` and using `const_cast` to force passing a const argument to a non const argument.

For example, `void Launch(const gd::EventsList& events) { Launch(const_cast<...>(events); };` to have the const version of Launch to launch the Non Const version (because we can guarantee it won't do any non const calls) and reimplement all the VisitXxx as non const version calling the new const version....

At the end it's probably simpler/safer to duplicate the code :)